### PR TITLE
chore(registry): update regex check for namespace

### PR DESCRIPTION
**Pull request by heiruwu**
Because

- We want to check if `namespace` is present in docker push URL
- Some requests might be missing auth `username`

This commit

- add back `username` empty string check
- add `namespace` section check in regex

### CODE REVIEW

File: plugins/registry/main.go [hunk 1]
```diff
@@ -108,7 +108,7 @@ func (r registerer) registerHandlers(ctx context.Context, extra map[string]inter
 		// [registry]/[namespace]/[repository path]:[image tag]
 		// The namespace is the user uid or the organization uid
 		var namespace string
-		matches := regexp.MustCompile(`/v2/([^/]+).*`).FindStringSubmatch(req.URL.Path)
+		matches := regexp.MustCompile(`/v2/([A-Za-z0-9\_]+)/([^/]+)/(blobs|manifests)/.*`).FindStringSubmatch(req.URL.Path)
```
Comment (heiruwu) [inline plugins/registry/main.go:111]: hi @pinglin @jvallesm not sure if you guys have better ways to check the existence of `namespace` here

Comment (jvallesm) [reply plugins/registry/main.go:111]: I'd use the same expression for the owner and the model ID. Right now this won't capture usernames with hyphens: [example](https://go.dev/play/p/0WBsNiDTt6S).

If we want to capture just 2 segments `{owner}/{imageID}`, then I'd copy the `[^/]+)` expression. If we want to capture an arbitrary number (I'm not sure about that, [complicates things](https://linear.app/instill-ai/issue/INS-3958/implement-registry-tag-list#comment-cbbd8d60) at the gateway routing level), then we can use the `+` operator.

Comment (heiruwu) [reply plugins/registry/main.go:111]: we actually do have regex rules for `namespace` and resource `ID`, but I can't seems to find it now. I will put both segment as `([^/]+)`, and we can re-evaluate when we encounter some supposedly unsupported name from users

Comment (heiruwu) [reply plugins/registry/main.go]: > If we want to capture an arbitrary number

quoted from @pinglin, I think we will limit the URL to just {namespace}/{artifact-id}
> I am also buying the way Replicate and other registry-based apps do more concise and user-friendly, i.e., having the image path [registry]/<namespace>/<artifact-id>. We can do whatever necessary image management in the artifact-backend. Also, the artifact-id should be self-explained enough to indicate it's a model, vdp, chart, etc., artifact.

Comment (jvallesm) [reply plugins/registry/main.go]: Oh I wasn't aware of these rules, in that case might make sense to replicate them here. But definitely hyphens are allowed, I had a peek and we have users with them.

Already merged but 👍 Looks good!

**Pull request merged**